mipengine/controller/api/DTOs/AlgorithmSpecificationsDTOs.py

```diff
@@ -0,0 +1,150 @@
+from dataclasses import dataclass
+from typing import List, Optional, Dict
+
+from dataclasses_json import dataclass_json
+
+from mipengine.controller.common.algorithms_specifications import CROSSVALIDATION_ALGORITHM_NAME, \
+    GenericParameterSpecification, AlgorithmSpecifications, AlgorithmsSpecifications
+from mipengine.controller.common.utils import Singleton
+
+INPUTDATA_PATHOLOGY_PARAMETER_NAME = "pathology"
+INPUTDATA_DATASET_PARAMETER_NAME = "dataset"
+INPUTDATA_FILTERS_PARAMETER_NAME = "filter"
+INPUTDATA_X_PARAMETER_NAME = "x"
+INPUTDATA_Y_PARAMETER_NAME = "y"
+
+
+@dataclass_json
+@dataclass
+class InputDataSpecificationDTO:
+    """
+    InputDataParameterDTO is different from the InputDataParameter
+    on the stattypes field.
+    It is optional on the DTOs, due to the dataset and pathology parameters.
+    """
+    label: str
+    desc: str
+    types: List[str]
+    notblank: bool
+    multiple: bool
+    stattypes: Optional[List[str]] = None
+    enumslen: Optional[int] = None
+
+
+def get_pathology_parameter():
+    return InputDataSpecificationDTO(
+        "Pathology of the data.",
+        "The pathology that the algorithm will run on.",
+        ["text"],
+        True,
+        False,
+        None,
+        None,
+    )
+
+
+def get_dataset_parameter():
+    return InputDataSpecificationDTO(
+        "Set of data to use.",
+        "The set of data to run the algorithm on.",
+        ["text"],
+        True,
+        True,
+        None,
+        None,
+    )
+
+
+def get_filter_parameter():
+    return InputDataSpecificationDTO(
+        "Filter on the data.",
+        "Features used in my algorithm.",
+        ["jsonObject"],
+        False,
+        False,
+        None,
+        None,
+    )
+
+
+@dataclass_json
+@dataclass
+class GenericParameterSpecificationDTO(GenericParameterSpecification):
+    """
+    GenericParameterDTO is identical to the GenericParameterSpecification
+    but exists for consistency and future use if needed.
```

Comment:
Why not introduce it in the future, if needed?

Reply:
On the api, the logic of DTOs is followed.
Whatever is transfered to and from the API has to be suffixed with "DTO".

The first approach was to use the same class but it just didn't feel right. All the classes should have the DTO suffix and be part of the DTOs package.

The second approach was to duplicate the class completely, to avoid breaking the UI when the algorithm parameters change. That didn't feel right as well because it was complete code duplication at this point.

A middle way was chosen to keep the DTO logic, have the DTOs all in the same place but also not duplicated code.